src/EA.Weee.RequestHandlers/Admin/CopyAatfHandler.cs

```diff
@@ -0,0 +1,89 @@
+﻿namespace EA.Weee.RequestHandlers.Admin
+{
+    using System;
+    using System.Threading.Tasks;
+    using Charges;
+    using Domain.Lookup;
+    using EA.Prsd.Core.Domain;
+    using EA.Prsd.Core.Mapper;
+    using EA.Prsd.Core.Mediator;
+    using EA.Weee.Core.AatfReturn;
+    using EA.Weee.Core.Helpers;
+    using EA.Weee.Domain.AatfReturn;
+    using EA.Weee.Domain.Organisation;
+    using EA.Weee.RequestHandlers.AatfReturn;
+    using EA.Weee.RequestHandlers.Security;
+    using EA.Weee.Requests.Admin;
+    using EA.Weee.Security;
+    using Shared;
+
+    public class CopyAatfHandler : IRequestHandler<CopyAatf, bool>
+    {
+        private readonly IWeeeAuthorization authorization;
+        private readonly IGenericDataAccess dataAccess;
+        private readonly IMap<AatfAddressData, AatfAddress> addressMapper;
+        private readonly IMap<AatfContactData, AatfContact> contactMapper;
+        private readonly ICommonDataAccess commonDataAccess;
+
+        public CopyAatfHandler(
+            IWeeeAuthorization authorization,
+            IGenericDataAccess dataAccess,
+            IMap<AatfAddressData, AatfAddress> addressMapper,
+            IMap<AatfContactData, AatfContact> contactMapper,
+            ICommonDataAccess commonDataAccess)
+        {
+            this.authorization = authorization;
+            this.addressMapper = addressMapper;
+            this.contactMapper = contactMapper;
+            this.commonDataAccess = commonDataAccess;
+            this.dataAccess = dataAccess;
+        }
+
+        public async Task<bool> HandleAsync(CopyAatf message)
+        {
+            authorization.EnsureCanAccessInternalArea();
+            authorization.EnsureUserInRole(Roles.InternalAdmin);
```

Comment:
How different is this handler to the AddAatf handler? Is it possible to re-use that handler with an additional parameter on the message to indicate if its a copy so that we can set the id?

Reply:
Could also add an integration test - to ensure the original AATF  for the year is untouched and the new one created with same aatf id

Reply:
Originally i did think but the copy will have more validations/checks added by further PBIs so decided to keep it separate.